node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -567,25 +665,86 @@ func ConvertScsiIdToNguid(scsiId string) string {
 	return finalNguid
 }
 
-func (o GetDmsPathHelperGeneric) WaitForDmToExist(volumeUuid string, volumeNguid string, maxRetries int, intervalSeconds int) (string, error) {
-	formatTemplate := strings.Join([]string{"%d", "%w"}, mpathdSeparator)
+func (o GetDmsPathHelperGeneric) GetMpathdOutput(volumeIdVariations []string,
+	multipathdCommandFormatArgs []string) (string, error) {
+	mpathdOutput, err := o.WaitForDmToExist(volumeIdVariations, WaitForMpathRetries,
+		WaitForMpathWaitIntervalSec, multipathdCommandFormatArgs)
+
+	if err != nil {
+		return "", err
+	}
+
+	if mpathdOutput == "" {
+		return "", &MultipathDeviceNotFoundForVolumeError{volumeIdVariations[0]}
+	}
+	return mpathdOutput, nil
+}
+
+func (o GetDmsPathHelperGeneric) WaitForDmToExist(volumeIdVariations []string, maxRetries int, intervalSeconds int,
+	multipathdCommandFormatArgs []string) (string, error) {
+	formatTemplate := strings.Join(multipathdCommandFormatArgs, mpathdSeparator)
 	args := []string{"show", "maps", "raw", "format", "\"", formatTemplate, "\""}
-	var err error
 	logger.Debugf("Waiting for dm to exist")
+	var err error
 	for i := 0; i < maxRetries; i++ {
 		err = nil
 		out, err := o.executer.ExecuteWithTimeout(TimeOutMultipathdCmd, multipathdCmd, args)
 		if err != nil {
 			return "", err
 		}
 		dms := string(out)
-		if !strings.Contains(dms, volumeUuid) && !strings.Contains(dms, volumeNguid) {
-			err = os.ErrNotExist
-		} else {
-			return dms, nil
+		for _, volumeIdVariation := range volumeIdVariations {
+			if strings.Contains(dms, volumeIdVariation) {
+				return dms, nil
+			}
 		}
+		err = os.ErrNotExist
 
 		time.Sleep(time.Second * time.Duration(intervalSeconds))
 	}
 	return "", err
 }
+
+func (o GetDmsPathHelperGeneric) ParseMpathName(volumeIdVariations []string, mpathdOutput string) map[string]bool {
+	dmFieldValues := make(map[string]bool)
+
+	scanner := bufio.NewScanner(strings.NewReader(mpathdOutput))
+	for scanner.Scan() {
+		volumeIdVariation, dmFieldValue := o.getLineParts(scanner)
```

Comment:
```suggestion
		volumeId, dmFieldValue := o.getLineParts(scanner)
```
we actually don't expect to get more than 1 variation from multipathd, so no need for "variation" here IMO

Reply:
done

Reply:
[no it isn't](https://github.com/IBM/ibm-block-csi-driver/pull/491#discussion_r887638476)